validation/src/blockchain/eth_rpc/types.rs

```diff
@@ -101,36 +93,25 @@ impl FromStr for EthProofType {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-/// A proof of time on the blockchain
-pub struct TimeProof {
+/// A proof of time derived from state on the blockchain
+pub struct TimeProofResults {
     /// The timestamp the proof was recorded
     pub timestamp: u64,
     /// The root CID of the proof
     pub root_cid: Cid,
 }
 
-/*
-    Planning to implemented this trait for Hoku using something like:
-
-    #[derive(Clone, Debug, PartialEq, Eq)]
-    pub struct HokuTxProof {
-        cid: Cid,
-        index: u64,
-    }
-*/
 #[async_trait::async_trait]
-/// Ethereum RPC provider methods. This is a higher level type than the actual RPC calls neeed and
+/// Wrapper around blockchain RPC provider that can be used to query blockchain state for the
+/// relevant information (such as the timestamp) for the given transaction described by an
+/// `AnchorProof`. This is a higher level type than the actual RPC calls need and
 /// may wrap a multiple calls into a logical behavior of getting necessary information.
 pub trait ChainInclusion {
-    /// The input format needed to do the inclusion proof
-    type InclusionInput;
-
     /// Get the CAIP2 chain ID supported by this RPC provider
     fn chain_id(&self) -> &caip2::ChainId;
 
     /// Get the block chain transaction if it exists with the block timestamp information
-    async fn chain_inclusion_proof(&self, input: &Self::InclusionInput)
-        -> Result<TimeProof, Error>;
+    async fn chain_inclusion_proof(&self, input: &AnchorProof) -> Result<TimeProofResults, Error>;
```

Comment:
this is the core of the change in this PR.  Now instead of having the `ChainInclusion` type by generic over an `InclusionInput` type that must be specified for each implementation of the trait, now we just simply take the raw `AnchorProof` object and let the implementations decide how to interpret that object.

This avoids the callers of this function needing to know whether the AnchorProof is for an ethereum anchor or an Adm anchor and having to build up the appropriate input type accordingly.  Now the caller can just blindly pass the anchor proof down into this function and trust the underlying implementation to do the right thing.

The goal is that other than the initial setup to construct the `ChainInclusionProviders` from the given RPC urls at startup, none of the event validation code will need to be in any way aware of which chain its providers are for, or have any chain-specific logic. All of that gets pushed down to the two implementations of the `ChainInclusion` trait.